### DISCUSSION TITLE
Supporting PHP Modifiers

### DISCUSSION THREAD

**Discussion by wblessen** [smarty-php/smarty#909] [General]
Upgrading to PHP 8.2 and Smarty 4.3 I see a lot of deprecation warnings in my usually clean log file:

Seems like PHP Functions as modifiers will be disabled and are marked as deprecated.

From the following thread I tried to implement the workaround:

 https://github.com/smarty-php/smarty/issues/813

``` php 
$this->smarty->addExtension(new WildcardExtension());
```

but unfortunatly, Smarty supports the function  addExtension just from Smarty5.0

I am loading Smarty with composer and do not get a stable Smarty 5

My questions:
Are their plans to release Smarty 5 as stable?
Are their other workarounds to keep the PHP Functions?

We have tons of them in our projects ....

**Comment by wblessen** [comment]
And I think, the easiest way to solve this could be, to enable PHP Modifier Support by Configuration.

Handling PHP Warnings and Errors by oneself, if PHP Modifiers are wanted.

**Comment by wisskid** [comment]
Yes, we are planning to release Smarty5 as stable soon.
Another workaround would be to implement handlers for all modifiers that you use.
Also, you can configure your application to not log deprecation errors?

**Reply by wblessen** [reply]
OK, I choosed the second option and  built modifiers for the PHP functions that we need.

Good luck with your release!
